### PR TITLE
Refactor/suggestions send all collection to backend

### PR DESCRIPTION
**Pull request by maria-j-k**
https://search-pr-758.docker-fid.grid.cyf-kr.edu.pl

### CODE REVIEW

**Comment by Marcelinna** [comment]
it is possible to display DATA SOURCE
image

**Comment by Marcelinna** [comment]
@michal-szostak 
all_collection = (
        "publication",
        "dataset",
        "software",
        "service",
        "data_source",
        "training",
        "guideline",
        "bundle",
    )

why we don't include providers and other ?

**Comment by Marcelinna** [comment]
@michal-szostak 
shouldn't I have the same results for the same collection when the data is the same in two different environments

image

File: ui/apps/ui/src/app/collections/services/fetch-data.service.ts [hunk 1]
```diff
@@ -52,13 +58,26 @@ export class FetchDataService {
       );
   }
 
+  fetchSuggestions$(
+    params: ISolrCollectionParams & ISolrSuggestionQueryParams
+  ): Observable<object> {
```
Comment (michal-szostak) [inline ui/apps/ui/src/app/collections/services/fetch-data.service.ts:63]: Better typing would be better here

File: ui/apps/ui/src/app/components/search-input/utils.ts [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
 
 export const toSuggestedResults = (
-  response: ISearchResults<IResult> & { link: string }
-): ISuggestedResults => ({
-  label: response.results[0]?.type.value,
-  results: response.results,
-  link: response.link,
-});
+  response: ISuggestionResponseResults<IResult>,
+
```
Comment (michal-szostak) [inline ui/apps/ui/src/app/components/search-input/utils.ts:6]: empty line here

**Review by michal-szostak — APPROVED** [review]

**Comment by eldenkat** [comment]
Inconsistent autocomplete:

1. In the collections: ALL and BUNDLES - there's **no** autosuggestion
2. In the collections: PUBLICATIONS, DATA, SOFTWARE and SERVICES autosuggestion appear after entering **2 letters.**
3. In the collections: DATA SOURCES, TRAINING ad IG -autosuggestion appear after entering **4 letters**.

**Comment by eldenkat** [comment]
No IG suggestion:
image

Zrzut ekranu 2023-08-09 084339

**Comment by eldenkat** [comment]
Trainings - autocomplete works only after 4th character:

Zrzut ekranu 2023-08-09 091930

Zrzut ekranu 2023-08-09 092031

Zrzut ekranu 2023-08-09 092101

Zrzut ekranu 2023-08-09 092127

**Comment by eldenkat** [comment]
Bundles - work only after 3 characters:

Zrzut ekranu 2023-08-09 095252

Zrzut ekranu 2023-08-09 095354

Zrzut ekranu 2023-08-09 095417

**Comment by eldenkat** [comment]
Autosuggstions work only for All Catalogs

**Comment by eldenkat** [comment]
1. Data sources - no autosuggestion until 4th character.
2. Trainings - no autosuggestion until 4th character.
3. Interoperability Guidelines - no autosuggestion until 5th character.
4. Bundles - no autosuggestion until 3rd character.

**Review by tomaszandel — APPROVED** [review]

**Review by agpul — APPROVED** [review]

**Pull request merged**